docs: add 'Deploying to Firebase Hosting' tutorial (#19960)
gatsbyjs/gatsby: docs/docs/deploying-to-firebase.md

---
title: Deploying to Firebase Hosting
---

In this guide, you will learn how to deploy your Gatsby site to Firebase Hosting.

> [Firebase Hosting](https://firebase.google.com/docs/hosting) is a free web content hosting for developers. With a single command, you can quickly deploy web apps and serve both static and dynamic content to a global CDN (content delivery network).

Reviewer: ```suggestion
[Firebase Hosting](https://firebase.google.com/docs/hosting) is a free web content hosting platform for developers. With a single command, you can quickly deploy web apps and serve both static and dynamic content to a global CDN (content delivery network).
```

## Pre-requisites

1. Make sure you have:

   - installed [Node.js](https://nodejs.org/en/download/) on your local machine.
   - a [Firebase Account](https://console.firebase.google.com).

1. This guide assumes that you already have a Gatsby project on your local machine.

## Deployment

1. Install the Firebase CLI with `npm` by running the following command:

   ```shell
   npm install -g firebase-tools
   ```

1. Sign into Firebase using your Google account by running the following command:

   ```shell
   firebase login
   ```

   You can test if the CLI is correctly installed by running `firebase projects:list`, which should show you a list of your Firebase Projects.

1. Navigate into your gatsby project directory and setup firebase:

   ```shell
   firebase init
   ```

   This command will prompt you to:

   - select the Firebase products you wish to setup. Be sure to select **Firebase Hosting**.

>    - select the Firebase Project you wish to use or create a new one, if you haven't done it previously.

Reviewer: Can we add some more information about creating a firebase project?

Author: Should we have it under this section? Or maybe have it on a different section, right in the beginning (prerequisites perhaps)?

Reviewer: Based on a quick search it looks as if there is a nice walk through after creating a Firebase account. So perhaps it's just a matter of editing the pre-reqs to note the creation of the account and a project.

Author: @laurieontech Thanks! I've added "Firebase Project" as a pre-req and added a link to the guide that explains how to create a project. Please take a look

>    When prompted to select your public directory, you can simply press <kbd>enter</kbd> since it defaults to `public`, which is also Gatsby's default public directory.

Reviewer: ```suggestion
   When prompted to select your public directory, press <kbd>enter</kbd>. It will default to `public`, which is also Gatsby's default public directory.
```
